src/main/java/io/confluent/kafkarest/AvroConsumerState.java

```diff
@@ -35,35 +38,42 @@
 
   // Note that this could be a static variable and shared, but that causes tests to break in
   // subtle ways because it causes state to be shared across tests, but only for the consumer.
-  private Decoder<Object> decoder = null;
+  private Deserializer<Object> deserializer = null;
 
-  public AvroConsumerState(KafkaRestConfig config,
-                           ConsumerInstanceId instanceId,
-                           ConsumerConnector consumer) {
-    super(config, instanceId, consumer);
-    Properties props = new Properties();
-    props.setProperty("schema.registry.url",
-                      config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
-    decoder = new KafkaAvroDecoder(new VerifiableProperties(props));
+  public AvroConsumerState(KafkaRestConfig config, ConsumerInstanceId instanceId,
+                           Properties consumerProperties,
+                           ConsumerManager.ConsumerFactory consumerFactory) {
+    super(config, instanceId, consumerProperties, consumerFactory);
+  }
+
+  private Deserializer<Object> initDeserializer() {
```

Comment:
In case of constructor initialization the superclass constructor is invoked before subclass and we get an NPE when getValue/KeyDeserializer() are called in superclass.